### PR TITLE
Making TextPainter rounding hack disabled by default

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Migrate tests in flutter/flutter. Once the tests here and in `*_customer_testing` are migrated, the default value of the migration flag will be changed from false to true, making the rounding hack disabled by default.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM :+1:.  Looks like a lot of cruft going away because of this rounding thing.  A super editor test is failing but I guess it has to be migrated too.

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -513,7 +513,11 @@ class TextPainter {
        _locale = locale,
        _strutStyle = strutStyle,
        _textWidthBasis = textWidthBasis,
-       _textHeightBehavior = textHeightBehavior;
+       _textHeightBehavior = textHeightBehavior,
+       assert(() {
+         ui.ParagraphBuilder.setDisableRoundingHack(true);
+         return true;
+       }());
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/painting/text_painter.dart:520]: Yeah it will be removed once https://github.com/flutter/engine/pull/44544 lands.

**Review by HansMuller — APPROVED** [review]

**Comment by LongCatIsLooong** [comment]
Changed dev/bots/test.dart so the flag gets enabled only on CI for now. Will remove that once https://github.com/flutter/engine/pull/44544 lands.

**Comment by LongCatIsLooong** [comment]
Setting google testing to passing: the `pr_test` script failed for me and this change shouldn't affect google3.

**Pull request merged**